src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.util.platforms.windows;
+
+import com.aws.greengrass.util.platforms.Exec;
+
+import java.lang.reflect.Method;
+import java.nio.file.Path;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+@SuppressWarnings("PMD.AvoidCatchingThrowable")
+public class WindowsExec extends Exec {
+    @Nullable
+    @Override
+    public Path which(String fn) {
+        // TODO
+        return null;
+    }
+
+    @Override
+    protected Process createProcess(String[] command) {
+        WindowsPlatform.WindowsRunasUserOptions userOptions =
+                (WindowsPlatform.WindowsRunasUserOptions) this.userOptions;
+        WindowsRunasProcess process = new WindowsRunasProcess(userOptions.getDomain(), userOptions.getUser(),
+                userOptions.getPassword());
+        process.setLpEnvironment(computeEnvironmentBlock());
+        process.setLpCurrentDirectory(dir.getAbsolutePath());
+        process.start(String.join(" ", command));
+        return process;
+    }
+
+    /**
+     * Convert environment Map to lpEnvironment block format.
+     * @return environment block for starting a process
+     */
+    private String computeEnvironmentBlock() {
+        // TODO: is reflection good here or should we copy over the source code
```

Comment:
if the source isn't bad, let's copy it over. No guarantee that this isn't changed in future JDKs